### PR TITLE
fossid-webapp: Make the timeout for FossID connections configurable

### PR DESCRIPTION
**Pull request by nnobelis**
When ORT waits for FossID scan to complete, the current timeout is
hardcoded and is set to one hour.
We have some FossID scans lasting for longer therefore the timeout needs
to be configurable.

Signed-off-by: Nicolas Nobelis <nicolas.nobelis@bosch.io>

### CODE REVIEW

File: scanner/src/main/kotlin/scanners/fossid/FossIdConfig.kt
Comment (sschuberth) [inline scanner/src/main/kotlin/scanners/fossid/FossIdConfig.kt:124]: Broken formatting.

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: scanner/src/main/kotlin/scanners/fossid/FossId.kt
Comment (oheger-bosch) [inline scanner/src/main/kotlin/scanners/fossid/FossId.kt:572]: Should the download use the same timeout? I would assume that this would be much faster than a scan operation.

File: model/src/main/resources/reference.conf
Comment (oheger-bosch) [inline model/src/main/resources/reference.conf:120]: I am confused what this value actually means and how it is related to the maximum amount of time we wait for a scan. For usability, wouldn't it be better to specify the timeout directly? The property would then have to be named accordingly.

**Comment by nnobelis** [comment]
Do not merge please: I got an idea last evening. I will change the impl.

File: model/src/main/resources/reference.conf
Comment (nnobelis) [reply model/src/main/resources/reference.conf:120]: This is exactly my new idea. I will change the implementation.

File: scanner/src/main/kotlin/scanners/fossid/FossId.kt
Comment (nnobelis) [reply scanner/src/main/kotlin/scanners/fossid/FossId.kt:572]: You're right. However don't you think keeping only one timeout lower the complexity of the code ?

File: scanner/src/main/kotlin/scanners/fossid/FossId.kt
Comment (oheger-bosch) [inline scanner/src/main/kotlin/scanners/fossid/FossId.kt:572]: What does this line do?

File: model/src/main/resources/reference.conf
Comment (oheger-bosch) [inline model/src/main/resources/reference.conf:120]: As the timeout is a rather big value (in the range of minutes up to an hour), I would suggest using a different time unit than milliseconds. Maybe minutes would be more appropriate. WDYT?

File: scanner/src/main/kotlin/scanners/fossid/FossId.kt
Comment (nnobelis) [reply scanner/src/main/kotlin/scanners/fossid/FossId.kt:572]: Copy/Paste error :)

File: model/src/main/resources/reference.conf
Comment (nnobelis) [reply model/src/main/resources/reference.conf]: Yes this is a good idea. I would go for minutes too.

File: scanner/src/main/kotlin/scanners/fossid/FossIdConfig.kt
Comment (oheger-bosch) [inline scanner/src/main/kotlin/scanners/fossid/FossIdConfig.kt:84]: Minor nit: As we now use minutes as unit for timeouts, this property should probably better of type `Int` rather than `Long`, since Java APIs typically use `Long` values to represent milliseconds. But I leave this to you.

**Review by oheger-bosch — APPROVED** [review]

**Review by oheger-bosch — DISMISSED** [review]

File: scanner/src/test/kotlin/scanners/fossid/FossIdConfigTest.kt
Comment (sschuberth) [inline scanner/src/test/kotlin/scanners/fossid/FossIdConfigTest.kt:55]: Is the timeout now really in minutes, so this is 300 minutes?

**Review by oheger-bosch — APPROVED** [review]

File: scanner/src/test/kotlin/scanners/fossid/FossIdConfigTest.kt
Comment (nnobelis) [reply scanner/src/test/kotlin/scanners/fossid/FossIdConfigTest.kt:55]: The timeout is now in minutes so 300 minutes -> 5 hours.
Or did I understood the question wrong ?

Comment (sschuberth) [reply scanner/src/test/kotlin/scanners/fossid/FossIdConfigTest.kt:55]: Yes, I was just wondering about that huge timeout.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**
